go/repository/document.go

```diff
@@ -0,0 +1,39 @@
+package repository
+
+import (
+	"challenge-flutter-go/models"
+
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type DocumentRepository struct {
+	Database *gorm.DB
+}
+
+// Get a document by its ID
+func (d *DocumentRepository) Get(id string) (document models.Document, err error) {
+	err = d.Database.First(&document, id).Error
+	return
+}
+
+// Create a document & associate it with a trip
+func (d *DocumentRepository) Create(document *models.Document) error {
+	return d.Database.Create(&document).Preload(clause.Associations).First(&document).Error
+}
+
+// Get all the documents of a trip
+func (d *DocumentRepository) GetDocuments(trip models.Trip) (documents []models.Document, err error) {
+	err = d.Database.Model(&models.Document{}).Where("trip_id = ?", trip.ID).Find(&documents).Error
+	return
+}
+
+// Delete a document from a trip
+func (d *DocumentRepository) DeleteDocument(trip models.Trip, documentID uint) (err error) {
+	var document models.Document
+	// il faut load les details du doc pour le hook, sinon le dans le hook le document est empty
+	if err = d.Database.Where("id = ?", documentID).First(&document).Error; err != nil {
+		return err
+	}
+	return d.Database.Delete(&document).Error
```

Comment:
```suggestion
	result := d.Database.Delete(&models.document{}, documentID)
	if result.RowsAffected == 0 {
		return gorm.ErrRecordNotFound
	}
	return result.Error
```

T'as un exemple pour la suppression d'un transport ici https://github.com/ESGI-69/challenge-flutter-go/blob/4f96bc4db010d6503da69f3f4ca7c843bb827d97/go/repository/transport.go#L23-L30

Comme ça, qu'**une seule query** sur la base de donnée pour une suppression. À la place de 2, une pour récupérer le document et une pour le supprimer.

Reply:
le hook recupere pas le doc meme avec ca